unified_planning/model/types.py

```diff
@@ -75,6 +79,17 @@ def is_bool_type(self) -> bool:
         return True
 
 
+class _DerivedBoolType(Type):
```

Comment:
Does this information needs to be in the typing system or we can keep a list/map/set of predicates that are derived in the `Problem`?

Reply:
In the beginning, I simply wanted to use the Boolean type, but I ran into some problems. For example, when a planning problem is created, an assertion checks whether all variables have an initial value. For derived predicates, however, we cannot set such a value. 

I think your suggestion of having a list/set of variables that are derived would also work if we can access it at the appropriate places. It will definitely simplify the code as the typing system is not touched. I will check it.